File: src/pocketcloud/cloud/terminal/log/level/CloudLogLevel.php
<?php

namespace pocketcloud\cloud\terminal\log\level;

use pocketcloud\cloud\util\enum\EnumTrait;

/**
 * @method static CloudLogLevel INFO()
 * @method static CloudLogLevel WARN()
 * @method static CloudLogLevel ERROR()
 * @method static CloudLogLevel DEBUG()
 * @method static CloudLogLevel SUCCESS()
 */
final class CloudLogLevel {
    use EnumTrait;

    protected static function init(): void {
        self::register("info", new CloudLogLevel("INFO", "§bINFO"));
        self::register("warn", new CloudLogLevel("WARN", "§cWARN"));
        self::register("error", new CloudLogLevel("ERROR", "§4ERROR"));
        self::register("success", new CloudLogLevel("SUCCESS", "§aSUCCESS"));
        self::register("debug", new CloudLogLevel("DEBUG", "§6DEBUG"));
    }

    public function __construct(
        private readonly string $name,
        private readonly string $prefix
    ) {}

    public function getName(): string {
        return $this->name;
    }

    public function getPrefix(): string {
        return $this->prefix;
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/console/log/CloudLogLevel.java
package de.pocketcloud.cloud.console.log;

import de.pocketcloud.network.packet.type.LogType;
import lombok.Getter;
import lombok.experimental.Accessors;

@Getter
@Accessors(fluent = true)
public enum CloudLogLevel {

    INFO("§bINFO"),
    WARN("§cWARN"),
    ERROR("§4ERROR"),
    SUCCESS("§aSUCCESS"),
    DEBUG("§6DEBUG");

    private final String prefix;

    CloudLogLevel(String prefix) {
        this.prefix = prefix;
    }

    public static CloudLogLevel toLogLevel(LogType type) {
        return switch (type) {
            case INFO -> CloudLogLevel.INFO;
            case WARN -> CloudLogLevel.WARN;
            case ERROR -> CloudLogLevel.ERROR;
            case SUCCESS -> CloudLogLevel.SUCCESS;
            case DEBUG -> CloudLogLevel.DEBUG;
        };
    }
}